src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateTemporaryCertificateIntegrationTest.kt

```diff
@@ -133,4 +152,125 @@ internal class GenerateTemporaryCertificateIntegrationTest : IntegrationTest() {
+
+        // When
+        val response = webTestClient.post()
+            .uri(URI_TEMPLATE, ERO_ID)
+            .bearerToken(getBearerToken(eroId = ERO_ID, groups = listOf("ero-$ERO_ID", "ero-vc-admin-$ERO_ID")))
+            .contentType(MediaType.APPLICATION_JSON)
+            .withBody(requestBody)
+            .exchange()
+            .expectStatus()
+            .isBadRequest
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        assertThat(actual)
+            .hasStatus(400)
+            .hasError("Bad Request")
+            .hasMessageContaining("Temporary Certificate gssCode '$GSS_CODE' is not valid for eroId '$ERO_ID'")
+    }
+
+    @Test
+    fun `should return temporary certificate pdf given valid request for authorised user`() {
+        // Given
+        val photoLocationArn = addPhotoToS3()
+        val request = buildGenerateTemporaryCertificateRequest(photoLocation = photoLocationArn)
+        val localAuthorityName = aValidLocalAuthorityName()
+        val localAuthorities: List<LocalAuthorityResponse> = listOf(
+            buildLocalAuthorityResponse(
+                gssCode = request.gssCode,
+                contactDetailsEnglish = buildContactDetails(name = localAuthorityName)
+            )
+        )
+        val eroResponse = buildElectoralRegistrationOfficeResponse(id = ERO_ID, localAuthorities = localAuthorities)
+        wireMockService.stubCognitoJwtIssuerResponse()
+        wireMockService.stubEroManagementGetEroByGssCode(eroResponse, request.gssCode)
+
+        // When
+        val response = webTestClient.mutate()
+            .codecs { it.defaultCodecs().maxInMemorySize(MAX_SIZE_1_MB) }
+            .build().post()
+            .uri(URI_TEMPLATE, ERO_ID)
+            .bearerToken(getBearerToken(eroId = ERO_ID, groups = listOf("ero-$ERO_ID", "ero-vc-admin-$ERO_ID")))
+            .contentType(MediaType.APPLICATION_JSON)
+            .withBody(request)
+            .exchange()
+            .expectStatus().isCreated
+            .expectHeader().contentType(MediaType.APPLICATION_PDF)
+            .expectBody(ByteArray::class.java)
+            .returnResult()
+
+        // Then
+        val temporaryCertificates = temporaryCertificateRepository.findByGssCodeInAndSourceTypeAndSourceReference(
+            listOf(request.gssCode),
+            VOTER_CARD,
+            request.sourceReference
+        )
+        assertThat(temporaryCertificates).hasSize(1)
+        val temporaryCertificate = temporaryCertificates[0]
+        val contentDisposition = response.responseHeaders.contentDisposition
+        assertThat(contentDisposition.filename)
+            .isEqualTo("temporary-certificate-${temporaryCertificate.certificateNumber}.pdf")
+
+        val pdfContent = response.responseBody
+        assertThat(pdfContent).isNotNull
+        PdfReader(pdfContent).use { reader ->
```

Comment:
This assertion on the field will fail with my [PR](https://github.com/cabinetoffice/eip-ero-print-api/pull/140) because I flatten the PDF to remove the fields

Reply:
I have a separate task to update to use the PDF templates provided by Malcolm - will DM you.